src/plots/barPlot.ts

```diff
@@ -499,21 +523,20 @@ export class Bar<X, Y> extends XYPlot<X, Y> {
     return intersected;
   }
 
-  private _updateValueScale() {
+  private _updateLengthScale() {
     if (!this._projectorsReady()) {
       return;
     }
-    const valueScale = this._isVertical ? this.y().scale : this.x().scale;
-    if (valueScale instanceof QuantitativeScale) {
-      const qscale = <QuantitativeScale<any>> valueScale;
-      qscale.addPaddingExceptionsProvider(this._baselineValueProvider);
-      qscale.addIncludedValuesProvider(this._baselineValueProvider);
+    const lengthScale = this.length().scale;
+    if (lengthScale instanceof QuantitativeScale) {
+      lengthScale.addPaddingExceptionsProvider(this._baselineValueProvider);
+      lengthScale.addIncludedValuesProvider(this._baselineValueProvider);
     }
   }
 
   protected _additionalPaint(time: number) {
-    const primaryScale: Scale<any, number> = this._isVertical ? this.y().scale : this.x().scale;
-    const scaledBaseline = primaryScale.scale(this.baselineValue());
+    const lengthScale = this.length().scale;
+    const scaledBaseline = lengthScale.scale(this.baselineValue());
```

Comment:
I actually found different places in the code had different interpretations of primary/secondary - some parts thought primary meant "x for vertical, y for horizontal" while other parts thought primary meant the opposite.

Independent/dependent could work too; I just felt length/thickness had more of a visual feel to it